components/homme/src/share/cxx/HybridVCoord.cpp

```diff
@@ -221,11 +221,14 @@ void HybridVCoord::compute_eta ()
   auto l_hybrid_bm = hybrid_bm;
   auto l_hybrid_ai = hybrid_ai;
   auto l_hybrid_bi = hybrid_bi;
+  const auto l_ps0 = ps0;
+  const auto p0 = PhysicalConstants::p0;
+  const auto kappa = PhysicalConstants::kappa;
 
   Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace>(0,NUM_LEV),
                        KOKKOS_LAMBDA(const int& ilev){
     l_etam(ilev) = l_hybrid_am(ilev) + l_hybrid_bm(ilev);
-    l_exner0(ilev) = pow(l_etam(ilev)*ps0/PhysicalConstants::p0,PhysicalConstants::kappa);
+    l_exner0(ilev) = pow(l_etam(ilev)*l_ps0/p0,kappa);
```

Comment:
these seems to be triggered randomly, another one is in HyperviscosityFunctorImpl.cpp file, Luca fixed it for a100 builds, there are probably more in the code.

Reply:
Agreed. Using a host constant should always fail, but in practice it depends on how the code is compiled, so sometimes these misuses stick around for a while.

Separately, and to be clear, this modification has nothing to do with the main content of the PR; I was just fixing something in Hommexx I found when running cuda-memcheck.